addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -9389,6 +9389,61 @@ QUnit.module("Views", (hooks) => {
         );
     });
 
+    QUnit.test(
+        'buttons with "confirm-title" and "confirm-label" attributes save before calling the method',
+        async function (assert) {
+            const actionService = {
+                start() {
+                    return {
+                        async doActionButton(args) {
+                            assert.step("execute_action");
+                        },
+                    };
+                },
+            };
+            registry.category("services").add("action", actionService, { force: true });
+
+            await makeView({
+                type: "form",
+                resModel: "partner",
+                serverData,
+                arch: `
+                    <form>
+                        <header>
+                            <button name="post" class="p" string="Confirm" type="object" confirm="Very dangerous. U sure?"
+                                confirm-title="Confirmation" confirm-label="Confirm"/>
+                        </header>
+                        <sheet>
+                            <field name="foo"/>
+                        </sheet>
+                    </form>`,
+                mockRPC(route, args) {
+                    assert.step(args.method);
+                },
+            });
+
+            // click on button, and cancel in confirm dialog
+            await click(target.querySelector(".o_statusbar_buttons button"));
+            assert.ok(
+                target.querySelector(".o_statusbar_buttons button").disabled,
+                "button should be disabled"
+            );
+            await click(target.querySelector(".modal-footer button.btn-secondary"));
+            assert.notOk(
+                target.querySelector(".o_statusbar_buttons button").disabled,
+                "button should no longer be disabled"
+            );
+
+            assert.verifySteps(["get_views", "onchange"]);
+
+            // click on button, and click on ok in confirm dialog
+            await click(target.querySelector(".o_statusbar_buttons button"));
+            assert.verifySteps([]);
+            await click(target.querySelector(".modal-footer button.btn-primary"));
+            assert.verifySteps(["create", "read", "execute_action"]);
```

Comment:
I think this part is not necessary, because it's already tested bellow, I would just check that the title / label is correct in the modal 

also, can we move it just below the original test ? (as it's a special case)